packages/utils/crates/utils-types/src/primitives.rs

```diff
@@ -1,211 +1,460 @@
+/// Represents a type of the balance: native or HOPR tokens.
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+pub enum BalanceType {
+    Native,
+    HOPR
+}
+
+/// Represents balance of some coin or token.
+#[derive(Clone, Debug)]
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
 pub struct Balance {
     value: u256,
+    balance_type: BalanceType
 }
 
-impl BaseBalance for Balance {
-    const SYMBOL: &'static str = "mHOPR";
-
-    fn value(&self) -> &u256 {
-        &self.value
+impl PartialEq for Balance {
+    fn eq(&self, other: &Self) -> bool {
+        self.eq(other)
     }
 }
 
-impl TypeBase for Balance {
-    fn to_hex(&self) -> String { hex::encode(self.value().to_be_bytes()) }
-}
+#[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen)]
+impl Balance {
+    #[cfg_attr(feature = "wasm", wasm_bindgen::prelude::wasm_bindgen(constructor))]
+    pub fn from_str(value: &str, balance_type: BalanceType) -> Self {
+        Balance {
+            value: u256::from_str_radix(value, 10).unwrap(),
+            balance_type,
+        }
+    }
 
-impl PartialEq for Balance {
-    fn eq(&self, other: &Self) -> bool {
-        assert_eq!(self.symbol(), other.symbol());
-        self.value().eq(other.value())
+    /// Retrieves the type (symbol) of the balance
+    pub fn balance_type(&self) -> BalanceType {
+        self.balance_type
     }
-}
 
-impl ToString for Balance {
-    fn to_string(&self) -> String {
+    pub fn to_hex(&self) -> String { hex::encode(self.value().to_be_bytes()) }
+
+    pub fn to_string(&self) -> String {
         self.value.to_string()
     }
-}
 
-impl Balance {
-    pub const ZERO: Self = Self { value: u256::ZERO };
+    pub fn eq(&self, other: &Balance) -> bool {
+        assert_eq!(self.balance_type(), other.balance_type());
+        self.value().eq(other.value())
+    }
 
-    pub fn from_str(value: &str) -> Result<Self, String> {
-        Ok(Balance {
-            value: u256::from_str_radix(value, 10).map_err(|_| "failed to parse")?,
-        })
+    /// Serializes just the value of the balance (not the symbol)
+    pub fn serialize_value(&self) -> Box<[u8]> {
+        self.value().to_be_bytes().into()
+    }
+
+    pub fn lt(&self, other: &Balance) -> bool {
+        assert_eq!(self.balance_type(), other.balance_type());
+        self.value().lt(other.value())
+    }
+
+    pub fn lte(&self, other: &Balance) -> bool {
+        assert_eq!(self.balance_type(), other.balance_type());
+        self.value().lt(other.value()) || self.value().eq(other.value())
+    }
+
+    pub fn gt(&self, other: &Balance) -> bool {
+        assert_eq!(self.balance_type(), other.balance_type());
+        self.value().gt(other.value())
+    }
+
+    pub fn gte(&self, other: &Balance) -> bool {
+        assert_eq!(self.balance_type(), other.balance_type());
+        self.value().gt(other.value()) || self.value().eq(other.value())
     }
```

Comment:
This is where https://doc.rust-lang.org/stable/std/cmp/trait.PartialOrd.html trait would be better.